lib/livebook/k8s/pod.ex

```diff
@@ -0,0 +1,167 @@
+defmodule Livebook.K8s.Pod do
+  @main_container_name "livebook-runtime"
+  @home_pvc_volume_name "lb-home-folder"
+  @default_pod_template """
+  apiVersion: v1
+  kind: PodTemplate
```

Comment:
Sorry, tiny nitpick, but I don't think this needs to be a template, right? it can just be directly the pod configuration. It seems all we do with the template is to extract the template key anyway, so we might as well get rid of the indirection?

Reply:
Since this particular resource is never stored on the cluster, it would technically be okay to make it a Pod. 
I gave cluster-stored PodTemplate resources some more thought though. Maybe there is actually a use case for supporting them. E.g. if you want to have a set of different templates for different GPUs (have a template for a100 nodes, one for T4, etc.), you could create these resources on the cluster and then just select the correct one according to your needs. 
Or you think that's too far-fetched?

Reply:
I have updated the code. It now supports `Pod` and/or `PodTemplate` manifests. I have made the default config a `Pod` though as you are right, the extra metadata didn't really add any value.

Reply:
> It now supports Pod and/or PodTemplate manifests.

Honestly, let's support only Pod, unless someone really needs template. Because otherwise you will also have to support PodTemplates on `flame_k8s_backend` as well. Fewer options for now is better!